sympy/series/formal.py

```diff
@@ -1464,6 +1452,130 @@ def __rmul__(self, other):
         return self.__mul__(other)
 
 
+class FiniteFormalPowerSeries(FormalPowerSeries):
+    """Represents Finite Formal Power Series of a function.
+
+    No computation is performed. This class should only to be used to represent
+    a series. No checks are performed.
+
+    The difference between `FiniteFormalPowerSeries` and `FormalPowerSeries` is that
+    the coefficient sequence of a `FiniteFormalPowerSeries` is a list of coefficients,
+    instead of a sequence as in a `FormalPowerSeries` object.
+
+    `product`, `compose`, and `inverse` functions of `FormalPowerSeries` class, return
+    an object of this class.
+
+    See Also
+    ========
+
+    sympy.series.formal.FormalPowerSeries
+    sympy.series.formal.fps
+    """
+
+    def __init__(self, *args):
+        pass
+
+    @property
+    def infinite(self):
+        raise NotImplementedError("No infinite version for an object of"
+                     " FiniteFormalPowerSeries class.")
+
+    def _eval_terms(self, n):
+        pass
+
+    def _eval_term(self, pt):
+        raise NotImplementedError
+
+    def polynomial(self, n):
+        return self._eval_terms(n)
+
+    def _eval_derivative(self, x):
+        raise NotImplementedError
+
+    def integrate(self, x):
+        raise NotImplementedError
+
+
+class FormalPowerSeriesProduct(FiniteFormalPowerSeries):
+    def __new__(cls, *args):
+        args = map(sympify, args)
+        return FormalPowerSeries.__new__(cls, *args)
+
+    @property
+    def f(self):
+        return self.args[0][0]
+
+    @property
+    def g(self):
+        return self.args[0][1]
```

Comment:
You might want check if other Properties defined in ``FormalPowerSeries`` still make sense.